tests/integration/security/reachability_test.go

```diff
@@ -130,6 +131,12 @@ func TestReachability(t *testing.T) {
 							return false
 						}
 
+						// Exclude calls from naked->VM since VM is not injected with sidecar
```

Comment:
VM is not injected sidecar? Isn't it hard-coded to always have one? https://github.com/istio/istio/blob/365a94720d5a5290fafeaee85d49ad83f144be1e/pkg/test/framework/components/echo/kube/deployment.go#L218

Reply:
It will have a sidecar running using the .deb file, but this sidecar is not "injected". In practice, I tried to call VMs using `naked` and all connections have timeout errors. My theory to explain the timeout is that since `naked` instances are also not "injected" with sidecar, it does not know the IP of the VM, thus the failure. Please correct me if I am wrong since I am still new to Envoy proxy.

Reply:
The naked pod doesn't have sidecar so how does it know the address of the VM in this case? I'm not very familiar with the mesh expansion but I guess there is no k8s Service defined for the VM, which means there is no DNS record for the VM instance in this case.

If that is the cause and is expected, you can just update the comments here.

Reply:
Yes, I think you are right: there's no k8s Service defined for the VM, so the naked instance won't know the address of the VM, which is why I am excluding the calls from naked to VMs. Sorry for the confusing wording I've been using.

Reply:
Very close to correct - there is a k8s service but there is no associated endpoints (instead there is workload entry only). So right now naked to VM won't work

Reply:
Thank you so much for your clarification! @howardjohn

Reply:
Regarding to the comment "so the naked instance won't know the address of the VM", does the workload entry for the VM contain the endpoint addresses of the VM? If so, why the naked instance won't know the address of the VM? 
The example workload entry on https://istio.io/latest/blog/2020/workload-entry/ contains the endpoint addresses of the VM (i.e., "address: 1.1.1.1" in the example).

Reply:
Naked means no envoy, so kubernetes does all of the traffic routing. Kubernetes knows absolutely nothing about workload entry, so the fact one exists with the address is not relevant

Reply:
I see, the k8s service is created but the selector doesn't match the VM deployment, that's why there is no endpoint created, thanks for the explanation.

Reply:
Workload entry is a k8s CRD; a naked pod without Envoy can still query k8s for the workload entry of a VM to obtain the endpoint address of the VM. Therefore, I am not sure about the comment "so the naked instance won't know the address of the VM". Can you provide a more detailed explanation?

Reply:
lei, I see what you mean in theory but that's not how it works in practice. The pod sends a `curl vm` request, which kube-proxy routes to the correct endpoint. However kube proxy doesn't have any endpoints, because it doesn't read workload entries it only reads endpoints.

with envoy it does work because we program envoy to have the addresses in the workload entry

Reply:
John, thank you for the detailed explanation!